runtime/scripts/results.js

```diff
@@ -0,0 +1,211 @@
+/*
+Copyright 2022 Newcastle University
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+/** @file Functions related to the download and interpretation of student results, and interaction with the results page, usually for outside-LTI contexts. */
+Numbas.queueScript('results', ['jme'], function () {
+
+
+    /** Functions related to the download and interpretation of student results, and interaction with the results page, usually for outside-LTI contexts.
+     *
+     * @namespace Numbas.results */
+    var results = Numbas.results = /** @lends Numbas.results */ {
+        // items should be accessible through Numbas.results.function, so either write them inside this as key:function pairs, or if necessary as:
+        //var ensure_decimal = math.ensure_decimal = function(n) { ? We need them to be like this to ensure they're accessible from elsewhere, maybe.
+        
+        /** Ensures a string will not cause issues within a csv due to commas, quotes, etc. 
+         * 
+         * @param {string} cell 
+         * @returns {string}
+         */
+        escape_csv_cell : function (cell) {
+            cell = cell + '';
+            if(cell.match(/[,"']/)) {
+                cell = '"'+cell.replace(/\\/g,'\\\\').replace(/"/g,'\\"')+'"';
+            }
+            return cell;
+        },
+        
+        /** Escapes each cell of a list of strings such that each will not cause issues within a csv
+         * 
+         * @param {Array.<string>} cells 
+         * @returns {Array.<string>} 
+         */
+        make_csv_row : function (cells) {
+            return cells.map(Numbas.results.escape_csv_cell).join(',');

```

Comment:
since you've set `results = Numbas.results` at the top of the file, you can just write `results.escape_csv_cell` here